Fixing some problems

GrateBrittan opened the pull request:
Should fix ponies going into the ground (Though I am unsure) and fixes people from being stuck in tase mode when you tase one person and attach it to someone else while the other person is still being tased. Though please correct me if I am wrong

GrateBrittan committed:
Fixing some problems

Should fix ponies going into the ground (Though I am unsure) and fixes people from being stuck in tase mode when you tase one person and attach it to someone else while the other person is still being tased. Though please correct me if I am wrong

swampservers commented on lua/weapons/weapon_taser.lua:
```diff
@@ -24,7 +24,7 @@ if CLIENT then
 
         local originvec, originang = ply:GetBonePosition(1)
-        ply:ManipulateBonePosition(0, Vector(0, 0, -35))
+        ply:ManipulateBonePosition(0, Vector(0, 0, -math.abs(ply:GetViewOffset().z)))
```
For a human, this value is 64, for a pony, I think it's 42. Maybe you want `ply:IsPony() and -13 or -35`

chev2 commented on lua/weapons/weapon_taser.lua:
```diff
@@ -74,6 +74,7 @@ function SWEP:PrimaryAttack() -- attempt attach to a player
     local trc = self.Owner:GetEyeTrace()
 
     if IsValid(trc.Entity) and trc.Entity:IsPlayer() and self.Owner:GetPos():DistToSqr(trc.Entity:GetPos()) < self.MaxTaseDist*self.MaxTaseDist then
+        UnTasePlayer()
```
`UnTasePlayer()` won't work, use `self:UnTasePlayer()` instead

GrateBrittan replied:
Oops, sorry I didn't notice

GrateBrittan committed:
Updated fix

corrected some thing I forgot to add and added a check to make it so you can't get tased while sitting or in a protected area

swampservers commented:
Thanks!
